app/controllers/concerns/account_concern.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module AccountConcern
+  extend ActiveSupport::Concern
+
+  ACCOUNT_SESSION_HEADER_INTERNAL_NAME = "HTTP_GOVUK_ACCOUNT_SESSION"
+  ACCOUNT_SESSION_HEADER_NAME = "GOVUK-Account-Session"
+  ACCOUNT_END_SESSION_HEADER_NAME = "GOVUK-Account-End-Session"
+  ACCOUNT_SESSION_DEV_COOKIE_NAME = "govuk_account_session"
+
+  included do
+    before_action :fetch_account_session_header
+    before_action :set_account_session_header
+    before_action :set_account_variant
+
+    helper_method :logged_in?
+
+    attr_accessor :account_session_header
+  end
+
+  def logged_in?
+    account_session_header.present?
+  end
+
+  def fetch_account_session_header
+    @account_session_header =
+      if request.headers[ACCOUNT_SESSION_HEADER_INTERNAL_NAME]
```

Comment:
We've not tested this path and probably should.
@barrucadu can I find out a bit more about the use case here?

I wonder if we can simplify the block at all if this is no longer needed?

Reply:
The issue I had in finder-frontend is that, with the controller tests, the custom header shows up as `HTTP_GOVUK_ACCOUNT_SESSION`, but with the feature specs it shows up as `GOVUK-Account-Session`.  Or maybe it was the other way around.

The only way I could get both sets of tests passing is by checking for both header names, which is a bit unfortunate.

Reply:
Ok I have simplifed this as we don't seem to be getting the feature test problem here.

Reply:
Can we add a feature spec, or something else which goes through the full request stack, exercising this code?  It would be bad to write a feature which only works in controller tests and breaks when proper HTTP requests are involved.

Reply:
I think so, yes.  Those at least support a `headers:` field on the request methods.